pkg/connect/state/state.go

```diff
@@ -268,15 +274,18 @@ func (g *WorkerGroup) Sync(ctx context.Context, groupManager WorkerGroupManager,
 	}
 
 	// Update the worker group to make sure it store the appropriate IDs
-	if syncReply.IsSuccess() {
-		g.AppID = syncReply.AppID
-		g.SyncID = syncReply.SyncID
-
-		// Update the worker group with the syncID so it's aware that it's already sync'd before
-		// Always update the worker group for consistency, even if the context is cancelled
-		if err := groupManager.UpdateWorkerGroup(context.Background(), g.EnvID, g); err != nil {
-			return fmt.Errorf("error updating worker group: %w", err)
-		}
+	if !syncReply.IsSuccess() {
+		// We always expect the App ID & Sync ID to be included in a sync result, representing either the idempotent reply or the new sync.
+		return fmt.Errorf("invalid sync result")
+	}
+
+	g.AppID = syncReply.AppID
```

Comment:
This is the crucial line for routing messages. If this is not set, the next UpsertConnection call will not create the proper indexes in the metadata store and messages will fail to route to the connection because it can't be identified as responsible for the app.